doc/neps/nep-0029-deprecation_policy.rst

```diff
@@ -0,0 +1,190 @@
+``Python`` cadence we avoid the edge cases where a project releases
+the month after ``Python`` and would effectively only support one
+version of ``Python`` that has an install base (and then the window
+is 42).  This buffer also makes us resilient to minor fluctuations /
+delays in the ``Python`` release schedule.
+
+Because the threshold for dropping a version of ``Python`` is based
+on historical release dates and a project's plans, the decision to drop
+support for a given version of ``Python`` can be made very early in
+the release process.
+
+There will be some unavoidable mismatch in supported version of
+``Python`` between packages immediately after a version of
+``Python`` ages out.  However this should not last longer than one
+release cycle of each of the projects, and when a given project
+does a minor or major release, it is guaranteed that there will be a
+stable release of all other projects that support the set of
+``Python`` the new release will support.
+
+If there is a major version bump in Python this policy will have to be
+re-addressed as we can not predict how drastic the changes will be and
+how the community will react.
+
+
+Implementation
+--------------
+
+We suggest that all projects adopt the following language into their
+development guidelines:
+
+
+   - support minor versions of ``Python`` initially released
+     42 months prior to our planned release date
+   - always support at least the 2 latest minor versions of ``Python``
+   - support minor versions of ``numpy`` initially released in the 24
+     months prior to our planned release date or oldest that supports the
+     minimum Python version (whichever is higher)
+
+   The minimum supported version of ``Python`` will be set to
+   ``python_requires`` in ``setup`` and all supported versions of
+   Python will be in the test matrix and have binary artifacts built
+   for releases.
+
+   We will bump the minimum Python and NumPy versions as we can on
+   every minor and major release, but never on a patch release.
+
+For other dependencies, adopt similar time windows of 24 months or
+shorter.
+
+
+Backward compatibility
+----------------------
+
+No issues other than the intentional dropping of old version of
+``Python`` and dependencies.
+
+
+Alternatives
+------------
+
+Ad-Hoc
+~~~~~~
+
+Projects could on every release evaluate if they want to increase
+the minimum version of Python supported.  While this is a notionally
+simple policy, it makes it hard for downstream users to predict what
+the future minimum versions will be.  As there is no objective threshold
+to when the minimum version should be dropped, it is easy for these
+discussions to devolve into [bike shedding](https://en.wikipedia.org/wiki/Wikipedia:Avoid_Parkinson%27s_bicycle-shed_effect) and acrimony.
+
+
+All Python Software Foundation supported versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is a very clear and conservative approach.  However, it means that
```

Comment:
The CPython supported versions of Python are listed in the Python Developers Guide and the Python PEPs. Supporting these are a very clear...